src/Parsers/ParserSelectQuery.cpp

```diff
@@ -125,7 +125,11 @@ bool ParserSelectQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     {
         bool has_all = false;
         if (!s_select.ignore(pos, expected))
-            return false;
+        {
+            /// This allows queries without SELECT, like `1 + 2`.
+            if (!implicit_select || with_expression_list || tables)
```

Comment:
I don't understand why the condition contains `with_expression_list || tables`

Reply:
With WITH statement, queries without SELECT became ugly and ambiguous, that's why we want to forbid them.

If a query has FROM, it's better to be conservative for now and still require SELECT.